.circleci/config.yml

```diff
@@ -78,37 +31,88 @@ jobs:
           # For example sqlalchemy requires greenlet only on certain platforms
           name: Freeze requirements
           command: pip-compile --allow-unsafe ~/project/requirements/requirements.in > ~/project/requirements/requirements.txt
+  do-static-analysis:
+    description: Do linting and other static analysis
+    steps:
       - run:
           name: Linting
           command: python3 -m flake8 --max-line-length=90
       - run:
           name: MyPy
           command: mypy --version && mypy sematic
+  non-coverage-tests:
+    description: Do tests without tracking code coverage
+    steps:
       - run:
           name: Run Non-coverage Tests
           # This assumes pytest is installed via the install-package step above
           command: PYTHONUNBUFFERED=1 bazel test //sematic/... --test_output=all
+  coverage-tests:
+    description: Do tests with coverage and upload coverage results
+    steps:
+      # try running coverage up to 4 times in a row. Weirdly, Circle doesn't
+      # have a formal retry mechanism:
+      # https://support.circleci.com/hc/en-us/articles/360043188514-How-to-Retry-a-Failed-Step-with-when-Attribute-
       - run:
-          name: Run Coverage Tests
+          name: Set Up Coverage Tests
           # This assumes pytest is installed via the install-package step above
-          command: PYTHONUNBUFFERED=1 bazel coverage //sematic/... --test_output=all --combined_report=lcov --test_tag_filters=cov
+          command: echo "bazel coverage //sematic/... --test_output=all --combined_report=lcov --test_tag_filters=cov" > ./do_coverage && chmod +x ./do_coverage
+      - run:
+          name: Run Coverage Tests
+          command: PYTHONUNBUFFERED=1 bash do_coverage
+      - run:
+          when: on_fail
+          name: Retry Coverage Tests (1)
+          command: PYTHONUNBUFFERED=1 bash do_coverage
+      - run:
+          when: on_fail
+          name: Retry Coverage Tests (2)
+          command: PYTHONUNBUFFERED=1 bash do_coverage
+      - run:
+          when: on_fail
+          name: Retry Coverage Tests (3)
+          command: PYTHONUNBUFFERED=1 bash do_coverage
```

Comment:
Is there no native way to do a retry?

Reply:
no, weirdly. I put a comment about that above (line 55)